next/components/forms/segments/AccountSections/MyApplicationsSection/MyApplicationsSection.tsx

```diff
@@ -1,18 +1,70 @@
 import MyApplicationsHeader from 'components/forms/segments/AccountSections/MyApplicationsSection/MyApplicationsHeader'
-import { useGlobalStateContext } from 'components/forms/states/GlobalState'
+import { useRouter } from 'next/router'
 import { useTranslation } from 'next-i18next'
+import { useEffect, useState } from 'react'
+import { useEffectOnceWhen } from 'rooks'
 
 import MyApplicationsList from './MyApplicationsList'
 
+const sections = ['SENT', 'SENDING', 'DRAFT'] as const
+export type ApplicationsListVariant = (typeof sections)[number]
+
+const isValidSection = (hash: string): hash is ApplicationsListVariant => {
+  return (sections as readonly string[]).includes(hash)
+}
+
+const useCurrentSection = () => {
+  const [currentSection, setCurrentSection] = useState<ApplicationsListVariant>('SENT')
+  const router = useRouter()
+
+  useEffectOnceWhen(() => {
+    // If the URL contains a valid hash, use it to set the currentSection
+    if (window.location.hash && isValidSection(window.location.hash.slice(1))) {
+      setCurrentSection(window.location.hash.slice(1) as ApplicationsListVariant)
+      return
+    }
+
+    // Otherwise, sync the hash with the currentSection
+    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+    router.push(`#${currentSection}`)
```

Comment:
while router.push failing is and edgecase of and edgecase, it's still nicer to handle the potential error instead of silencing it

```suggestion
    router.push(`#${currentSection}`).catch(error => logger.error(error))
```

you may need to import logger - it's an instance of a tool which let's us send frontend errors to our backend monitoring, so that we'll atleast know if something fishy starts happening